include/pybind11/cast.h

```diff
@@ -34,6 +34,20 @@ template <typename Type> class opaque {
 
 NAMESPACE_BEGIN(detail)
 
+/// Convert from any type pointer to another avoding warnings about void* vs (void*()) incompatibility.
```

Comment:
Tried it on Ubuntu 14.04 with 

```
c++ (Ubuntu 4.8.4-2ubuntu1~14.04.1) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

 `__GNUC_PREREQ` wasn't defined, and trying without guard gave me

```
warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
 #    pragma GCC diagnostic ignored "-Wpendantic"
```